### PR TITLE
Clean up Optional-related code.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Remove `SimplifyOptionalCallChains`, as the `Optional#isEmpty()` method is now available in the Android SDK starting with version 33 and is desugared.
- Make use of `Optional` method chaining.

#### Fixes the following issue(s)

- Fixes #

#### APK testing

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Stypox** [comment]
Why do you think `SimplifyOptionalCallChains` is not needed anymore? It was not added by accident: it was added because the IDE was complaining. So why does Android Studio say it's not a valid method, but at the same time suggests it? Is it desugared properly?

image
image

File: app/src/main/java/org/schabi/newpipe/player/playback/MediaSourceManager.java [hunk 1]
```diff
@@ -422,21 +421,24 @@ private void maybeLoadItem(@NonNull final PlayQueueItem item) {
...
-                        FailedMediaSource.of(stream, new MediaSourceResolutionException(message));
-            }
-
-            final MediaItemTag tag = MediaItemTag.from(source.getMediaItem()).get();
-            final long expiration = System.currentTimeMillis()
-                    + ServiceHelper.getCacheExpirationMillis(streamInfo.getServiceId());
-            return new LoadedMediaSource(source, tag, stream, expiration);
+            final var source = playbackListener.sourceOf(stream, streamInfo);
+
+            return Optional.ofNullable(source)
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/player/playback/MediaSourceManager.java:426]: And also turn `source1` into `source` below

```suggestion
            return Optional.ofNullable(playbackListener.sourceOf(stream, streamInfo))
```

File: app/src/main/java/org/schabi/newpipe/player/Player.java [hunk 2]
```diff
@@ -2036,40 +2030,35 @@ public void useVideoSource(final boolean videoEnabled) {
...
-        }
+                trackSelector.setParameters(parametersBuilder);
+            }
 
-        setRecovery();
+            setRecovery();
+        }, () -> {
+            reloadPlayQueueManager();
+            setRecovery();
+        });
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/player/Player.java:2061]: Maybe add a comment in this lambda to make it clear that it is called when we have no current stream info

**Comment by Isira-Seneviratne** [comment]
> Why do you think `SimplifyOptionalCallChains` is not needed anymore? It was not added by accident: it was added because the IDE was complaining. So why does Android Studio say it's not a valid method, but at the same time suggests it? Is it desugared properly?
> 
> image image

Yeah, the `isEmpty()` method was added to the Android SDK starting with version 33, and it gets desugared.

**Review by Stypox — CHANGES_REQUESTED** [review]
Other simplifications look good to me

**Comment by Stypox** [comment]
> Yeah, the isEmpty() method was added to the Android SDK starting with version 33, and it gets desugared.

Oh great! I had gradle out of sync with the ide

File: app/src/main/java/org/schabi/newpipe/player/mediaitem/MediaItemTag.java [hunk 3]
```diff
@@ -61,12 +61,11 @@ default Optional<Quality> getMaybeQuality() {
...
     static Optional<MediaItemTag> from(@Nullable final MediaItem mediaItem) {
-        if (mediaItem == null || mediaItem.localConfiguration == null
-                || !(mediaItem.localConfiguration.tag instanceof MediaItemTag)) {
-            return Optional.empty();
-        }
-
-        return Optional.of((MediaItemTag) mediaItem.localConfiguration.tag);
+        return Optional.ofNullable(mediaItem)
+                .map(item -> item.localConfiguration)
+                .map(localConfiguration -> localConfiguration.tag)
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/player/mediaitem/MediaItemTag.java]: `map` will crash if the return value of the lambda is `null`. I looked it up but there does not seem to be a `mapNullable` (just like `ofNullable` exists). So I used `flatMap(... -> Optional.ofNullable(...))`

```java
jshell> Optional.of(new Object()).map(null)
|  Exception java.lang.NullPointerException
|        at Objects.requireNonNull (Objects.java:233)
|        at Optional.map (Optional.java:256)
|        at (#2:1)

jshell> Optional.of(new Object()).flatMap(o -> Optional.ofNullable(null))
$4 ==> Optional.empty
```

File: app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java [hunk 4]
```diff
@@ -1887,10 +1883,8 @@ public void onServiceStopped() {
     @Override
     public void onFullscreenStateChanged(final boolean fullscreen) {
         setupBrightness();
-        //noinspection SimplifyOptionalCallChains
-        if (!isPlayerAndPlayerServiceAvailable()
-                || !player.UIs().get(MainPlayerUi.class).isPresent()
-                || getRoot().map(View::getParent).orElse(null) == null) {
+        if (!isPlayerAndPlayerServiceAvailable() || player.UIs().get(MainPlayerUi.class).isEmpty()
+                || getRoot().map(View::getParent).isEmpty()) {
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java]: Your code is perfectly equivalent, but I think the `.map(View::getParent)` was wrongly being used to allow `getParent` to return `null`.

**Review by Stypox — APPROVED** [review]
I made some small changes, in part described below, in the last commit, to fix some things and make further improvements. Thank you!

File: app/src/main/java/org/schabi/newpipe/player/mediaitem/MediaItemTag.java (hunk 3 above)
Comment (Isira-Seneviratne) [reply app/src/main/java/org/schabi/newpipe/player/mediaitem/MediaItemTag.java]: That's weird, `null` values should be converted to empty `Optional`s: https://developer.android.com/reference/java/util/Optional#map(java.util.function.Function%3C?%20super%20T,%20?%20extends%20U%3E)

Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/player/mediaitem/MediaItemTag.java]: I will revert that part of my commit

Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/player/mediaitem/MediaItemTag.java]: I pushed ad605e2c5a38e57af1b0ec7d4c991a9a9240dfff

**Pull request merged**